packages/core/src/modules/credentials/__tests__/CredentialService.test.ts

```diff
@@ -955,4 +955,64 @@ describe('CredentialService', () => {
       expect(result).toEqual(expect.arrayContaining(expected))
     })
   })
+
+  describe('declineOffer', () => {
+    const threadId = 'fd9c5ddb-ec11-4acd-bc32-540736249754'
+    let credential: CredentialRecord
+
+    beforeEach(() => {
+      credential = mockCredentialRecord({
+        state: CredentialState.OfferReceived,
+        tags: { threadId },
+      })
+    })
+
+    test(`updates state to ${CredentialState.OfferDeclinedSent}`, async () => {
+      // given
+      const repositoryUpdateSpy = jest.spyOn(credentialRepository, 'update')
+
+      // when
+      await credentialService.declineOffer(credential)
+
+      // then
+      expect(repositoryUpdateSpy).toHaveBeenCalledTimes(1)
+      const [[updatedCredentialRecord]] = repositoryUpdateSpy.mock.calls
+      expect(updatedCredentialRecord).toMatchObject({
+        state: CredentialState.OfferDeclinedSent,
+      })
```

Comment:
I prefer to have this syntax:

```suggestion
	  expect(repositoryUpdateSpy).toHaveBeenNthCalledWith(1, {
        state: CredentialState.OfferDeclinedSent,
      })
```